### DISCUSSION TITLE
7alpha28 - SELECT COUNT Returns a string instead of a number/bigint

### DISCUSSION THREAD

**Discussion by arsonik** [sequelize/sequelize#16494] [Help & Questions]
Is it a configuration thing ? 
since I've updated from .26 to .28 

Sequelize is initialised with the same configuration 
I've found that #typeCast() is receiving a VAR_STRING 

this test doesn't work 
```ts
    it('should ', async () => {
        const conn = await mariadb.createConnection({
            host: 'localhost',
            port: 5506,
            user: 'root',
            database: 'admin',
        });
        const nativeResults = await conn.query('SELECT COUNT(1) AS n FROM user');
        const seqResults = await User.sequelize?.query('SELECT COUNT(1) AS n FROM user', { type: QueryTypes.SELECT });
        expect(nativeResults?.[0].n).toBeTypeOf('bigint'); // SUCCEED
        expect(seqResults?.[0].n).toBeTypeOf('bigint'); // FAILS "AssertionError: expected '70' to be type of 'bigint'"
    });
```

**Comment by ephys** [comment]
This is normal (for now): We have not implemented #10468 yet

**Reply by arsonik** [reply]
sorry in alpha 26 it returns a typeof "number" which is fine !
how do I get that "number" back in alpha 28 ?

**Comment by arsonik** [comment]
understood but how come it was working on 7 alpha 26 ?

**Reply by ephys** [reply]
alpha 27 implemented support for mariadb v3, so if you used v3 before the output was indeterminate as we had no official support for it yet

we'll implement bigint support but we'll roll it out for all dialects in an unified manner as soon as we can
